File: 0733-flood-fill/0733-flood-fill.java
class Solution {
    public int[][] floodFill(int[][] image, int sr, int sc, int color) {
        int row = image.length;
        int col = image[0].length;
        int startingColor = image[sr][sc];
        boolean visited[][] = new boolean[row][col];
         solve(image, row, col, sr, sc, color, startingColor, visited);
//         for(int i=0; i<row; i++) {
//             for(int j=0; j<col; j++) {
//                 // if(image[i][j] == startingColor) {
                   
            
//                 // } 
//             }
//         }
        return image;
    }
    
    public static void solve(int[][] a, int row, int col, int i, int j, int color, int start, boolean[][] visited) {
        if(i < 0 || j < 0 || i >= row || j >= col || visited[i][j] || a[i][j] != start) {
            return;
        }
        a[i][j] = color;
        visited[i][j] = true;
        solve(a, row, col, i+1, j, color, start, visited);
        solve(a, row, col, i, j+1, color, start, visited);
        solve(a, row, col, i, j-1, color, start, visited);

        solve(a, row, col, i-1, j, color, start, visited);

    }
}

File: 0733-flood-fill/0733-flood-fill.cpp
class Solution {
public:
    bool isSafe( int row, int col, vector<vector<int>>& image, vector<vector<int>> &visited, int pcolor){
        
        int m = image.size();
        int n = image[0].size();
        
       if( row>=0 && row<m && col>=0 && col<n && !visited[row][col]           && image[row][col] == pcolor ){
           return true;
       }
        else return false;
        
    }
    void dfs( int row, int col,vector<vector<int>>& image, vector<vector<int>> &visited, int &color, int &pcolor ){
        
        visited[row][col] = 1;
        image[row][col] = color;
        
//         UP
         if( isSafe(row-1,col,image,visited,pcolor)){
            dfs(row-1, col, image, visited, color,pcolor);
        }
        
        if( isSafe(row,col-1,image,visited,pcolor)){
            dfs(row, col-1, image, visited, color,pcolor);
        }
        if( isSafe(row+1,col,image,visited,pcolor)){
            dfs(row+1, col, image, visited, color,pcolor);
        }
        
        if( isSafe(row,col+1,image,visited, pcolor)){
            dfs(row, col+1, image, visited, color,pcolor);
        }
    }
    
    vector<vector<int>> floodFill(vector<vector<int>>& image, int sr, int sc, int color) {
        
        int m = image.size();
        int n = image[0].size();
        int pcolor = image[sr][sc];
        vector<vector<int>> visited(m, vector<int>(n,0));
        dfs(sr,sc,image,visited, color, pcolor);
        
        return image;
        
    }
};